meshtastic/tests/test_node_runtime_seturl.py

```
@@ -0,0 +1,1498 @@
+    @pytest.mark.unit
+    def test_parse_without_hash_raises_error(self) -> None:
+        """parse() with URL missing hash fragment raises error."""
+        url = "https://meshtastic.org/d/"
+
+        captured_msg: list[str] = []
+
+        def raise_error(msg: str) -> NoReturn:
+            captured_msg.append(msg)
+            raise ValueError(msg)
+
+        with pytest.raises(ValueError):
```

Ruff (0.15.6)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)